finish/app/src/main/java/org/tensorflow/lite/codelabs/textclassification/MainActivity.java

```diff
@@ -110,39 +114,25 @@ private void showResult(final String textToShow) {
   // TODO 2: Implement a method to download TFLite model from Firebase
   /** Download model from Firebase ML. */
   private synchronized void downloadModel(String modelName) {
-    final FirebaseCustomRemoteModel remoteModel =
-        new FirebaseCustomRemoteModel
-            .Builder(modelName)
-            .build();
-    FirebaseModelDownloadConditions conditions =
-        new FirebaseModelDownloadConditions.Builder()
+      CustomModelDownloadConditions conditions = new CustomModelDownloadConditions.Builder()
             .requireWifi()
             .build();
-    final FirebaseModelManager firebaseModelManager = FirebaseModelManager.getInstance();
-    firebaseModelManager
-        .download(remoteModel, conditions)
-        .continueWithTask(task ->
-            firebaseModelManager.getLatestModelFile(remoteModel)
-        )
-        .continueWith(executorService, (Continuation<File, Void>) task -> {
-          // Initialize a text classifier instance with the model
-          File modelFile = task.getResult();
-
-          // TODO 6: Initialize a TextClassifier with the downloaded model
-          textClassifier = NLClassifier.createFromFile(modelFile);
-
-          // Enable predict button
-          predictButton.setEnabled(true);
-          return null;
-        })
-        .addOnFailureListener(e -> {
-          Log.e(TAG, "Failed to download and initialize the model. ", e);
-          Toast.makeText(
-              MainActivity.this,
-              "Model download failed, please check your connection.",
-              Toast.LENGTH_LONG)
-              .show();
-          predictButton.setEnabled(false);
-        });
+      FirebaseModelDownloader.getInstance()
+              .getModel("sentiment_analysis", DownloadType.LOCAL_MODEL, conditions)
+              .addOnSuccessListener(model -> {
```

Comment:
Since we are using `addOnSuccessListener`, shouldn't we also use `addOnFailureListener`?